build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryDependenciesClient.java

```diff
@@ -217,12 +222,26 @@ private HttpResponse executeDownload(String artifactUrl, boolean isHead, Map<Str
     }
 
     public void setProperties(String urlPath, String props) throws IOException {
-        String url = ArtifactoryHttpClient.encodeUrl(urlPath + "?properties=" + props);
+        String url = ArtifactoryHttpClient.encodeUrl(urlPath + "?properties=");
+        // Use URLEncoder to allow special characters use
+        url += encodeProps(props);
         PreemptiveHttpClient client = httpClient.getHttpClient();
         HttpPut httpPut = new HttpPut(url);
         checkNoContent(client.execute(httpPut), "Failed to set properties to '" + urlPath + "'");
     }
 
+    private String encodeProps(String props) throws UnsupportedEncodingException {
```

Comment:
The code already includes a method which encodes properties (as matrix params).
The method is used by the maven and gradle upload flows, but there's a chance it is also used by generic upload with file spec  See the *encodePath* method in the *DeploymentUrlUtils* class.
One of the methods should probably replace the other, or the code should be merged, depending on the advantages of each API.